### PR TITLE
Refactor Button component test file to new format

### PR DESCRIPTION
**Pull request by SriHV**
### What is the context of this PR?

[139](https://github.com/orgs/ONSdigital/projects/21/views/1?pane=issue&itemId=72216585)

Refactored the button macro.spec file as per the new format.

### How to review this PR

Ask yourself....
- Do we need any additional tests?
- Are the tests descriptions clear?
- Does the order of the test document make sense?
- Is it internally consistent?
- Is it consistent with the [guidelines for test refactoring](https://confluence.ons.gov.uk/display/SDC/Testing+Refactor+Guidance)? 

run this test command to view the test results:
`yarn test --testNamePattern="FOR: button"`

### Checklist

This needs to be completed by the person raising the PR.

-   [x] I have selected the correct Assignee
-   [x] I have linked the correct Issue

### CODE REVIEW

File: src/components/button/_macro.spec.js [hunk 1]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-            }),
-        );
-
-        expect($('.ons-btn').hasClass('ons-btn--download')).toBe(true);
-    });
-
-    it('has `download` icon when `variants` contains "download"', () => {
-        const faker = templateFaker();
-        const iconsSpy = faker.spy('icon');
+describe('FOR: button', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:8]:
```suggestion
describe('FOR: Macro: Button', () => {
```

File: src/components/button/_macro.spec.js [hunk 2]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+        describe('WHEN: there are no params given', () => {
+            it('THEN: renders expected style classes', () => {
+                const $ = cheerio.load(renderComponent('button'));
+                expect($('.ons-btn .ons-btn__inner').length).toBe(1);
+            });
 
-        faker.renderComponent('button', {
-            url: 'http://example.com',
-            variants: 'download',
+            it('THEN; renders `button` element', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:16]:
```suggestion
            it('THEN; renders button element', () => {
```

File: src/components/button/_macro.spec.js [hunk 3]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-
-    it('has `loader` icon when `variants` contains "loader"', () => {
-        const faker = templateFaker();
-        const iconsSpy = faker.spy('icon');
-
-        faker.renderComponent('button', {
-            variants: 'loader',
+    describe('GIVEN: Params: id', () => {
+        describe('WHEN: id is given as parameter', () => {
+            it('THEN: renders the provided `id` attribute', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:39]:
```suggestion
            it('THEN: renders the provided id attribute', () => {
```

File: src/components/button/_macro.spec.js [hunk 4]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-        it('has the provided link', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    url: 'http://example.com',
-                }),
-            );
-
-            expect($('a').attr('href')).toBe('http://example.com');
-        });
+            it('THEN: renders an `a` element', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:290]:
```suggestion
            it('THEN: renders an <a> element', () => {
```

File: src/components/button/_macro.spec.js [hunk 5]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-            expect($('button').attr('name')).toBe('special-name');
+                expect($('button').attr('type')).toBe('button');
+            });
         });
     });
 
-    describe('mode: link', () => {
-        it('passes jest-axe checks', async () => {
+    describe('GIVEN: Params: link', () => {
+        describe('WHEN: link parameter is provided', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:276]:
```suggestion
    describe('GIVEN: Params: url', () => {
        describe('WHEN: url parameter is provided', () => {
```

File: src/components/button/_macro.spec.js [hunk 6]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-            const iconsSpy = faker.spy('icon');
+            it('THEN: renders expected style classes', () => {
+                expect($('a').hasClass('ons-btn')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-btn--link')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-submit-btn')).toBe(true);
+            });
 
-            faker.renderComponent('button', {
-                url: 'http://example.com',
+            it('THEN: contains the provided link', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:300]:
```suggestion
            it('THEN: the link points to the provided url', () => {
```

File: src/components/button/_macro.spec.js [hunk 7]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
+                expect($('a').attr('href')).toBe('http://example.com');
             });
 
-            expect(iconsSpy.occurrences[0].iconType).toBe('arrow-next');
-        });
+            it('THEN: contains arrow-next icon by default', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:304]:
```suggestion
            it('THEN: the link button contains arrow-next icon by default', () => {
```

File: src/components/button/_macro.spec.js [hunk 8]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-                renderComponent('button', {
-                    variants: 'download',
-                    removeDownloadAttribute: true,
-                }),
-            );
-
-            expect($('.ons-btn').attr('download')).toBeUndefined();
+    describe('GIVEN: params: removeDownloadattribute', () => {
+        describe('WHEN: variants contains download and removeDownloadAttribute is true', () => {
+            it('THEN: does not have the download attribute', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:373]:
```suggestion
            it('THEN: the button does not have the download attribute', () => {
```

File: src/components/button/_macro.spec.js [hunk 9]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-                    url: 'http://example.com',
-                    newWindow: true,
-                    newWindowDescription: 'custom opens in a new window text',
-                }),
-            );
-
-            expect($('.ons-btn__new-window-description').text()).toBe(' (custom opens in a new window text)');
+    describe('GIVEN: Params: additonal classes', () => {
+        describe('WHEN: classes parameter is provided', () => {
+            it('THENL renders additionally provided style classes', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:346]:
```suggestion
            it('THEN: the button has the additionally provided style classes', () => {
```

File: src/components/button/_macro.spec.js [hunk 10]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-        it('does not have the `download` attribute when `variants` contains "download" and `removeDownloadAttribute` is `true`', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    variants: 'download',
-                    removeDownloadAttribute: true,
-                }),
-            );
-
-            expect($('.ons-btn').attr('download')).toBeUndefined();
+    describe('GIVEN: params: removeDownloadattribute', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:371]:
```suggestion
    describe('GIVEN: Params: removeDownloadattribute', () => {
```

File: src/components/button/_macro.spec.js [hunk 11]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-        it('has the `download` attribute when `variants` contains "download"', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    variants: 'download',
-                }),
-            );
-
-            expect($('.ons-btn').attr('download')).toBeDefined();
+        describe('WHEN: innerclasses parameter is provided', () => {
+            it('THEN: renders additionally provided inner style classes', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:358]:
```suggestion
            it('THEN: the button inner has the additionally provided inner style classes', () => {
```

File: src/components/button/_macro.spec.js [hunk 12]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-        it('has a default new window description when `newWindow` is `true`', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    url: 'http://example.com',
-                    newWindow: true,
-                }),
-            );
-
-            expect($('.ons-btn__new-window-description').text()).toBe(' (opens in a new tab)');
+        describe('WHEN: link parameter is provided, newWindow is true and newWindowDescription is provided', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:330]:
```suggestion
        describe('WHEN: url parameter is provided, newWindow is true and newWindowDescription is provided', () => {
```

File: src/components/button/_macro.spec.js [hunk 13]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-        it('has the `button` role', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    url: 'http://example.com',
-                }),
-            );
-
-            expect($('.ons-btn').attr('role')).toBe('button');
+        describe('WHEN: link parameter is provided and newWindow is true', () => {
+            it('THEN: renders a default new window description', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:320]:
```suggestion
            it('THEN: has default new window description', () => {
```

File: src/components/button/_macro.spec.js [hunk 14]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    url: 'http://example.com',
-                    newWindow: true,
-                }),
-            );
-
-            expect($('.ons-btn__new-window-description').text()).toBe(' (opens in a new tab)');
+        describe('WHEN: link parameter is provided, newWindow is true and newWindowDescription is provided', () => {
+            it('THEN: contains a custom new window description', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:331]:
```suggestion
            it('THEN: has custom new window description', () => {
```

File: src/components/button/_macro.spec.js [hunk 15]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-
-        it('has the `button` role', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    url: 'http://example.com',
-                }),
-            );
-
-            expect($('.ons-btn').attr('role')).toBe('button');
+        describe('WHEN: link parameter is provided and newWindow is true', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:319]:
```suggestion
        describe('WHEN: url parameter is provided and newWindow is true', () => {
```

File: src/components/button/_macro.spec.js [hunk 16]
```diff
@@ -303,130 +282,104 @@ describe('macro: button', () => {
...
-            expect($('a').attr('href')).toBe('http://example.com');
-        });
+            it('THEN: renders an `a` element', () => {
+                expect($('a').length).toBe(1);
+            });
 
-        it('has `arrow-next` icon by default', () => {
-            const faker = templateFaker();
-            const iconsSpy = faker.spy('icon');
+            it('THEN: renders expected style classes', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:294]:
```suggestion
            it('THEN: has expected style classes', () => {
```

File: src/components/button/_macro.spec.js [hunk 17]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+                });
 
-            expect($('button').attr('type')).toBe('button');
+                expect(iconsSpy.occurrences[0].iconType).toBe('print');
+            });
         });
+    });
+    describe('GIVEN: Params: type', () => {
+        describe('WHEN: type parameter is provided', () => {
+            it('THEN: renders the provided type attribute', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:235]:
```suggestion
            it('THEN: has the provided type attribute', () => {
```

File: src/components/button/_macro.spec.js [hunk 18]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+                expect($('button').attr('type')).toBe('special-type');
+            });
 
-        it('has the provided `value` attribute', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    value: 'special-value',
-                }),
-            );
+            it('THEN: renders the provided type attribute if variant contains print', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:245]:
```suggestion
            it('THEN: has the provided type attribute if variant contains print', () => {
```

File: src/components/button/_macro.spec.js [hunk 19]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+                        variants: 'print',
+                    }),
+                );
 
-            expect($('button').attr('value')).toBe('special-value');
+                expect($('button').attr('type')).toBe('special-type');
+            });
         });
+        describe('WHEN: type parameter is not provided', () => {
+            it('THEN: defaults to being a submit button when type is not provided', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:257]:
```suggestion
            it('THEN: defaults to being a submit button', () => {
```
This is already in the when

File: src/components/button/_macro.spec.js [hunk 20]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-        it('has the provided `name` attribute', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    name: 'special-name',
-                }),
-            );
+                expect($('button').attr('type')).toBe('submit');
+            });
+
+            it('THEN: defaults to being a button if variants contains print', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:263]: the "if variants contains print" should be in a when

File: src/components/button/_macro.spec.js (hunk 18 above)
Comment (rmccar) [inline src/components/button/_macro.spec.js:245]: "if variant contains print" should be in a when

File: src/components/button/_macro.spec.js [hunk 21]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+                expect($('.ons-btn').hasClass('ons-u-d-no')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-print-btn')).toBe(true);
+            });
 
-        it('defaults to being a "submit" button when `type` is not provided', () => {
-            const $ = cheerio.load(renderComponent('button'));
-
-            expect($('button').attr('type')).toBe('submit');
-        });
+            it('THEN: renders print icon when variants contains print', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:220]:
```suggestion
            it('THEN: button has the print icon', () => {
```

"when variants contains print" is already in the above when

File: src/components/button/_macro.spec.js [hunk 22]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    type: 'special-type',
-                    variants: 'print',
-                }),
-            );
-
-            expect($('button').attr('type')).toBe('special-type');
+        describe('WHEN: variants contains timer', () => {
+            it('THEN: renders provided variant style classes', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:196]:
```suggestion
            it('THEN: the button has the timer classes', () => {
```
Theres no classes provided, these are automatically set

File: src/components/button/_macro.spec.js [hunk 23]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+                    renderComponent('button', {
+                        variants: 'loader',
+                    }),
+                );
+
+                expect($('.ons-btn').hasClass('ons-btn--loader')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-loader')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-submit-btn')).toBe(true);
+            });
+            it('THEN: loader icon when variants contains loader', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:184]:
```suggestion
            it('THEN: the button has the loader icon', () => {
```
Already in the above when

File: src/components/button/_macro.spec.js [hunk 24]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+                    renderComponent('button', {
+                        variants: 'download',
+                    }),
+                );
+
+                expect($('.ons-btn').attr('download')).toBeDefined();
+            });
         });
+        describe('WHEN: variants contains loader', () => {
+            it('THEN: renders provided variant style classes', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:173]:
```suggestion
            it('THEN: the button has the loader classes', () => {
```
No classes have been provided these are automatic

File: src/components/button/_macro.spec.js [hunk 25]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
     });
-
-    it('has `chevron` icon when `variants` contains "mobile"', () => {
-        const faker = templateFaker();
-        const iconsSpy = faker.spy('icon');
-
-        faker.renderComponent('button', {
-            variants: 'mobile',
+    describe('GIVEN: Params: attributes', () => {
+        describe('WHEN: attributes is given as parameter', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:50]:
```suggestion
        describe('WHEN: the attributes param is provided', () => {
```

File: src/components/button/_macro.spec.js [hunk 26]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-
-        expect($('.ons-btn').hasClass('ons-btn--download')).toBe(true);
-    });
-
-    it('has `download` icon when `variants` contains "download"', () => {
-        const faker = templateFaker();
-        const iconsSpy = faker.spy('icon');
+describe('FOR: button', () => {
+    describe('GIVEN: Params: no params', () => {
+        describe('WHEN: there are no params given', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:10]:
```suggestion
        describe('WHEN: there are no params provided', () => {
```

File: src/components/button/_macro.spec.js [hunk 27]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
     });
-
-    it('has `print` icon when `variants` contains "print"', () => {
-        const faker = templateFaker();
-        const iconsSpy = faker.spy('icon');
-
-        faker.renderComponent('button', {
-            url: 'http://example.com',
-            variants: 'print',
+    describe('GIVEN: Params: standard', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:22]:
```suggestion
    describe('GIVEN: Params: text, name and value', () => {
```

File: src/components/button/_macro.spec.js [hunk 28]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+        describe('WHEN: button component is set with standard params', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:23]:
```suggestion
        describe('WHEN: text, name and value params are provided', () => {
```

File: src/components/button/_macro.spec.js [hunk 29]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
     });
-
-    it('has `loader` icon when `variants` contains "loader"', () => {
-        const faker = templateFaker();
-        const iconsSpy = faker.spy('icon');
-
-        faker.renderComponent('button', {
-            variants: 'loader',
+    describe('GIVEN: Params: id', () => {
+        describe('WHEN: id is given as parameter', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:38]:
```suggestion
        describe('WHEN: the id param is provided', () => {
```

File: src/components/button/_macro.spec.js [hunk 30]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-        expect($('.ons-btn').hasClass('ons-btn--download')).toBe(true);
-    });
-
-    it('has `download` icon when `variants` contains "download"', () => {
-        const faker = templateFaker();
-        const iconsSpy = faker.spy('icon');
+describe('FOR: button', () => {
+    describe('GIVEN: Params: no params', () => {
+        describe('WHEN: there are no params given', () => {
+            it('THEN: renders expected style classes', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:11]:
```suggestion
            it('THEN: has expected style classes', () => {
```

File: src/components/button/_macro.spec.js [hunk 31]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-    it('has label text when `html` is provided', () => {
-        const $ = cheerio.load(
-            renderComponent('button', {
-                html: 'Click <strong>me</strong>!',
-            }),
-        );
-
-        expect($('.ons-btn__text').html()).toBe('Click <strong>me</strong>!');
+    describe('GIVEN: Params: text', () => {
+        describe('WHEN: text is given as parameter', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:66]:
```suggestion
        describe('WHEN: the text param is provided', () => {
```

File: src/components/button/_macro.spec.js [hunk 32]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+            it('THEN: renders label text', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:67]:
```suggestion
            it('THEN: the button has the label text', () => {
```

File: src/components/button/_macro.spec.js [hunk 33]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-        const $ = cheerio.load(
-            renderComponent('button', {
-                buttonContext: 'button context text',
-            }),
-        );
-
-        expect($('.ons-btn__context').text()).toBe('button context text');
+    describe('GIVEN: Params: buttonContext', () => {
+        describe('WHEN: buttonContext parameter is provided', () => {
+            it('THEN: renders button context text', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:79]:
```suggestion
            it('THEN: the button has the provided button context', () => {
```

File: src/components/button/_macro.spec.js [hunk 34]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-                text: 'Click me!',
-                iconPosition: 'before',
-                iconType: 'exit',
-            }),
-        );
-
-        expect($('.ons-icon + .ons-btn__text').text()).toBe('Click me!');
+    describe('GIVEN: Params: html', () => {
+        describe('WHEN: html parameter is provided', () => {
+            it('THEN: renders label text', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:91]:
```suggestion
            it('THEN: the button label has the provided html', () => {
```

File: src/components/button/_macro.spec.js [hunk 35]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+                        text: 'Click me!',
+                        iconPosition: 'before',
+                        iconType: 'exit',
+                    }),
+                );
+                expect($('.ons-icon + .ons-btn__text').text()).toBe('Click me!');
+            });
+        });
+        describe('WHEN: iconPosition is set to after for custom icon', () => {
+            it('THEN: renders custom icon before button text', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:115]:
```suggestion
            it('THEN: renders custom icon after button text', () => {
```

File: src/components/button/_macro.spec.js [hunk 36]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-                    name: 'example',
-                    value: 'example-value',
-                }),
-            );
-
-            const results = await axe($.html());
-            expect(results).toHaveNoViolations();
+    describe('GIVEN: Params: variants', () => {
+        describe('WHEN: variants are present', () => {
+            it('THEN: renders provided variant style classes', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:130]:
```suggestion
            it('THEN: the button has the expected variant classes', () => {
```

File: src/components/button/_macro.spec.js [hunk 37]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
-            const $ = cheerio.load(renderComponent('button'));
+                faker.renderComponent('button', {
+                    url: 'http://example.com',
+                    variants: 'download',
+                });
 
-            expect($('button').length).toBe(1);
+                expect(iconsSpy.occurrences[0].iconType).toBe('download');
+            });
+            it('THEN: renders the download attribute when variants contains download', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:162]:
```suggestion
            it('THEN: the button has the download attribute', () => {
```

File: src/components/button/_macro.spec.js [hunk 38]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+                        variants: ['variant-a', 'variant-b'],
+                    }),
+                );
+
+                expect($('.ons-btn').hasClass('ons-btn--variant-a')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-btn--variant-b')).toBe(true);
+            });
         });
+        describe('WHEN: variants contains download', () => {
+            it('THEN: renders provided variant style class', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:142]:
```suggestion
            it('THEN: the button has the download class', () => {
```

File: src/components/button/_macro.spec.js [hunk 39]
```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
...
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        url: 'http://example.com',
+                        variants: 'download',
+                    }),
+                );
+                expect($('.ons-btn').hasClass('ons-btn--download')).toBe(true);
+            });
+            it('THEN: renders download icon', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:151]:
```suggestion
            it('THEN: the button has the download icon', () => {
```

**Review by rmccar — CHANGES_REQUESTED** [review]
These are mainly just minor things but it would be good to be consistent with the way we are writing these tests. We need to make sure we use the correct names for params and the tests describe what they are testing

File: src/components/button/_macro.spec.js [hunk 40]
```diff
@@ -5,296 +5,277 @@ import * as cheerio from 'cheerio';
...
-        expect($('.ons-btn').hasClass('ons-btn--download')).toBe(true);
-    });
-
-    it('has `download` icon when `variants` contains "download"', () => {
-        const faker = templateFaker();
-        const iconsSpy = faker.spy('icon');
+describe('FOR: Macro: Button', () => {
+    describe('GIVEN: Params: no params', () => {
+        describe('WHEN: there are no params provided', () => {
+            it('THEN: has expected style classes', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:11]: All the thens in this file should be using "test" and not "it"

File: src/components/button/_macro.spec.js [hunk 41]
```diff
@@ -303,130 +284,104 @@ describe('macro: button', () => {
...
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    url: 'http://example.com',
-                    newWindow: true,
-                    newWindowDescription: 'custom opens in a new window text',
-                }),
-            );
-
-            expect($('.ons-btn__new-window-description').text()).toBe('(custom opens in a new window text)');
+    describe('GIVEN: Params: additonal classes', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:346]:
```suggestion
    describe('GIVEN: Params: additional classes', () => {
```

File: src/components/button/_macro.spec.js [hunk 42]
```diff
@@ -303,130 +284,104 @@ describe('macro: button', () => {
...
-
-        it('has the `download` attribute when `variants` contains "download"', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    variants: 'download',
-                }),
-            );
-
-            expect($('.ons-btn').attr('download')).toBeDefined();
+        describe('WHEN: innerclasses parameter is provided', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:359]:
```suggestion
        describe('WHEN: innerClasses parameter is provided', () => {
```

File: src/components/button/_macro.spec.js [hunk 43]
```diff
@@ -303,130 +284,104 @@ describe('macro: button', () => {
...
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    variants: 'download',
-                    removeDownloadAttribute: true,
-                }),
-            );
-
-            expect($('.ons-btn').attr('download')).toBeUndefined();
+    describe('GIVEN: Params: removeDownloadattribute', () => {
+        describe('WHEN: variants contains download and removeDownloadAttribute is true', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:374]:
```suggestion
        describe('WHEN: variants contains download and removeDownloadAttribute is true', () => {
```

File: src/components/button/_macro.spec.js [hunk 44]
```diff
@@ -303,130 +284,104 @@ describe('macro: button', () => {
...
-        it('does not have the `download` attribute when `variants` contains "download" and `removeDownloadAttribute` is `true`', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    variants: 'download',
-                    removeDownloadAttribute: true,
-                }),
-            );
-
-            expect($('.ons-btn').attr('download')).toBeUndefined();
+    describe('GIVEN: Params: removeDownloadattribute', () => {
```
Comment (rmccar) [inline src/components/button/_macro.spec.js:373]:
```suggestion
    describe('GIVEN: Params: removeDownloadAttribute', () => {
```

**Comment by rmccar** [comment]
We need tests for the following attributes to make sure they are being tested:
- name
- value
- iconType
- listeners

**Review by rmccar — APPROVED** [review]

**Review by rmccar — APPROVED** [review]

**Review by admilne — APPROVED** [review]

**Pull request merged**